### PR TITLE
Nuclear Bomb shows damage visually

### PR DESCRIPTION
**Pull request by Flaborized**
## About the PR

It makes the nuclear bomb look more broken as it takes more damage. Doesn't affect it during Spacemas cuz there's no sprites for the spacemas nuke being damaged

## Why's this needed? 

I made the sprites when I resprited the nuke a while ago, never got implemented, here it is!! Being implemented!! Should add some visual clarity to the bomb instead of requiring you to examine it.

## Changelog
does not need a changelog entry imo but feel free

### CODE REVIEW

File: code/obj/machinery/nuclearbomb.dm [hunk 1]
```diff
@@ -83,10 +85,16 @@
 			switch(src.health)
 				if(80 to 125)
 					boutput(usr, "<span style=\"color:red\">It is a little bit damaged.</span>")
+					if(!isitspacemas)
```
Comment (pali6) [inline code/obj/machinery/nuclearbomb.dm:88]: This only updates the sprite when someone examines the nuke. I'd move the icon state changing logic into either take_damage or process (but probably the former).

Comment (Flaborized) [reply code/obj/machinery/nuclearbomb.dm]: i fixesd this very silly mistake!!!!

File: code/obj/machinery/nuclearbomb.dm [hunk 2]
```diff
@@ -284,6 +286,16 @@
...
 	proc/take_damage(var/amount)
+		if(!isitspacemas)
+			switch(src.health)
+				if(80 to 125)
+					src.icon_state = "nuclearbomb1"
+				if(40 to 79)
+					if(!isitspacemas)
+						src.icon_state = "nuclearbomb2"
+				if(1 to 39)
+					if(!isitspacemas)
```
Comment (pali6) [inline code/obj/machinery/nuclearbomb.dm:297]: These inner ifs are redundant. Also the ranges technically don't cover all values (e.g. taking damage from 125 to 39.5 wouldn't update the sprite as 39.5 doesn't lie in any of the ranges). But I can just fix that after merging it or something.

File: code/obj/machinery/nuclearbomb.dm [hunk 3]
```diff
@@ -284,6 +286,16 @@
...
+						src.icon_state = "nuclearbomb3"
```
Comment (pali6) [inline code/obj/machinery/nuclearbomb.dm:298]:
```suggestion
			switch(src.health)
				if(80 to 125)
					src.icon_state = "nuclearbomb1"
				if(40 to 80)
					src.icon_state = "nuclearbomb2"
				if(1 to 40)
					src.icon_state = "nuclearbomb3"
```

**Pull request merged**